lib/server.rb

```diff
@@ -3,69 +3,89 @@
 require_relative './server/controllers/browse_controller'
 require_relative './server/controllers/manifests_controller'
 require_relative './server/controllers/plots_controller'
+require_relative './server/controllers/view_controller'
 
 class Timur
   class Server < Etna::Server
-    # root path
+    # Root path.
     get '/' do
       erb_view(:welcome)
     end
 
-    # welcome_controller.rb
     get 'no_auth' do
       erb_view(:no_auth)
     end
 
-    with auth: { user: { can_view?: :project_name } } do
-      # activity_controller.rb
-      get ':project_name/activity', action: 'browse#activity', as: :activity
+    with auth: {user: {can_view?: :project_name}} do
 
-      # browse_controller.rb
-      get ':project_name', action: 'browse#index', as: :project
-      get ':project_name/browse', action: 'browse#index', as: :browse
-      get ':project_name/view/:model_name', action: 'browse#view', as: :view
+# PAGES. Delivers the base HTML and UI.
 
-      # browse view
+      # Browse page.
       get ':project_name/browse/:model_name/*record_name', as: :browse_model do
         erb_view(:model)
       end
 
-      # search view
+      # Search page.
       get ':project_name/search', as: :search do
         erb_view(:search)
       end
 
-      # map view
+      # Map page.
       get ':project_name/map', as: :map do
         erb_view(:map)
       end
 
+      # Settings page.
+      get ':project_name/settings/:settings_page', as: :settings do
+        erb_view(:settings)
+      end
+
+      # Manifest page.
+      get ':project_name/manifests', as: :manifests do
+        erb_view(:manifests)
+      end
+
+      # Plots page.
+      get ':project_name/plots', as: :plots do
+        erb_view(:plots)
+      end
+
+# CONTROLLER ACTIONS. Supports the API and backend calls.
+
+      # view_controller.rb
+      get ':project_name/view/:model_name',
+        action: 'view#view_json',
+        as: :view_json
+      post ':project_name/view/update_view_json',
```

Comment:
It is redundant to call these _json, all of the APIs deal in json. I'd go for /view/update and /view/delete

Reply:
+1